sdk/core/Azure.Core/src/Shared/Autorest/ArmOperationHelpers.cs

```diff
@@ -0,0 +1,356 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Helper methods for ARM long-running operations.
+    /// </summary>
+    internal static class ArmOperationHelpers
+    {
+        internal static Operation<Response> Create(HttpPipeline pipeline, ClientDiagnostics clientDiagnostics, Response originalResponse, RequestMethod requestMethod, string scopeName, OperationFinalStateVia finalStateVia, Func<HttpMessage> createOriginalHttpMessage)
```

Comment:
Are client devs expected to call these, or only generated code? Concerned that we don't even assert arguments if the former.

Reply:
Only generated code ATM.